### PR TITLE
Remove R1Tensor

### PR DESCRIPTION
**Pull request by CusiniM**
Also: https://github.com/GEOSX/LvArray/pull/205

### CODE REVIEW

**Comment by klevzoff** [comment]
@rrsettgast How do we want to proceed with functions like
```c++
R1Tensor LinePlaneIntersection( R1Tensor lineDir,
                                R1Tensor linePoint,
                                R1Tensor planeNormal,
                                R1Tensor planeOrigin );
```
in absence of a vocabulary type for tensors?

1. One option is to unconditionally convert everything to c-array references:
```c++
void LinePlaneIntersection( real64 const ( & lineDir )[ 3 ],
                            real64 const ( & linePoint )[ 3 ],
                            real64 const ( & planeNormal )[ 3 ],
                            real64 const ( & planeOrigin )[ 3 ],
                            real64 ( &result )[ 3 ] );
```
Syntax ugliness aside, this prevents directly supplying slices as arguments, potentially leading to the following use patterns:
```c++
real64 const lineDir[3] = LVARRAY_TENSOROPS_INIT_LOCAL_3( lineDirections[k] );
real64 const linePoint[3] = LVARRAY_TENSOROPS_INIT_LOCAL_3( linePoints[k] );
real64 const planeNormal[3] = LVARRAY_TENSOROPS_INIT_LOCAL_3( planeNormals[n] );
real64 const placeOrigin[3] = LVARRAY_TENSOROPS_INIT_LOCAL_3( placeOrigins[n] );
real64 intersection[3];
LinePlaneIntersection( lineDir, linePoint, planeNormal, placeOrigin, intersection );
LvArray::tensorOps::copy< 3 >( intersectionPoints[m], intersection );
```
This does a lot of unnecessary copies, increases register pressure (if the code snippet is part of a kernel) and is not super readable. This is kind of a worst-case scenario example, where all arguments and result are stored in some global arrays - in practice some arguments might already be local variables; nevertheless it's instructive to imagine some extreme use patterns.

2. Another approach is to template everything on array types, leading to something like
```c++
template< typename A1, typename A2, typename A3, typename A4, typename A5 >
void LinePlaneIntersection( A1 const && lineDir,
                            A2 const && linePoint,
                            A3 const && planeNormal,
                            A4 const && planeOrigin,
                            A5 && result );
```
This works to some extent, although requires us to put everything in headers, putting extra pressure on compile times (possibly minor effect). This does not work with virtual functions, like `PartitionBase::IsCoordInPartition()`. Of course one might argue that we shouldn't be calling virtual functions on individual points... but we have this in code now and need to deal with it. Of course, this signature is not very self-documenting... but that's the trade-off for generality. Also important to note that `const` parameters here do not enforce actual const-correctness (i.e. you may get an `arraySlice1d< real64 > const & lineDir` and accidentally write into it).

3. Finally, we could go case by case and provide specific overloads only for cases we need based on use patterns. In this case, it would be
```c++
void LinePlaneIntersection( real64 const ( & lineDir )[ 3 ],
                        arraySlice1d< real64 const > const linePoint,
                        real64 const ( & planeNormal )[ 3 ],
                        real64 const ( & planeOrigin )[ 3 ],
                        real64 ( &result )[ 3 ] );
```
Whenever someone needs to use it in a different context, where, say `planeOrigin` argument is a slice of a 2D array, they would have either to "materialize" it into a `real64[3]` local variable first, or add a different overload of `LinePlaneIntersection`. I _really_ don't like this approach, but listing it for completeness.

Opinions?

**Comment by rrsettgast** [comment]
@klevzoff I think option 2. We could still put it in the source file if we instantiate the templates. Depends on how many permutations we want to support. I think if there are a lot, then we just put it in the header.

**Comment by rrsettgast** [comment]
@klevzoff @CusiniM 
Or we could do something like:
```
template< typename A1, typename A2, typename A3, typename A4, typename A5 >
void LinePlaneIntersection_Impl( A1 const && lineDir,
                                 A2 const && linePoint,
                                 A3 const && planeNormal,
                                 A4 const && planeOrigin,
                                 A5 && result );
```
declared in the header, defined in the source.

and then
```
void LinePlaneIntersection( real64 const ( & lineDir )[ 3 ],
                            arraySlice1d< real64 const > const linePoint,
                            real64 const ( & planeNormal )[ 3 ],
                            real64 const ( & planeOrigin )[ 3 ],
                            real64 ( &result )[ 3 ] )
{
LinePlaneIntersection_Impl( lineDir, linePoint, planeNormal, planeOrigin, result );
}
```
declared in the header, defined in the source. Of course if you want this to be inline, everything has to be in the header.

File: src/coreComponents/physicsSolvers/fluidFlow/ProppantTransportKernels.cpp [hunk 1]
```diff
@@ -1166,7 +1168,7 @@ ProppantPackVolumeKernel::
                              real64 const proppantDensity,
                              real64 const proppantDiameter,
                              real64 const maxProppantConcentration,
-                             R1Tensor const unitGravityVector,
+                             R1Tensor const & unitGravityVector,
```
Comment (rrsettgast) [inline src/coreComponents/physicsSolvers/fluidFlow/ProppantTransportKernels.cpp:1171]: Is this a problem?

**Review by rrsettgast — APPROVED** [review]

File: src/coreComponents/finiteVolume/TwoPointFluxApproximation.cpp [hunk 2]
```diff
@@ -464,8 +471,19 @@ void TwoPointFluxApproximation::addToFractureStencil( MeshLevel & mesh,
...
+#if SET_CREATION_DISPLACEMENT==1
+                              , aperture
+                              , faceMap
+                              , faceNormal
+                              , faceToNodesMap
+                              , &touchedNodes
+                              , incrementalDisplacement
+                              , totalDisplacement
+                              , this
+#endif
```
[the remark this answers is not in the document]
Comment (klevzoff) [reply src/coreComponents/finiteVolume/TwoPointFluxApproximation.cpp:485]: All of these are used if `SET_CREATION_DISPLACEMENT` is set to 1 (but we don't test this configuration anywhere)

**Pull request merged**